airflow/www/views.py

```diff
@@ -2903,6 +2903,7 @@ def duration(self, dag_id, session=None):
             min_date = timezone.utc_epoch()
         ti_fails = (
             session.query(TaskFail)
+            .options(joinedload(TaskFail.task_instance))
```

Comment:
We might want this on `gantt` as well, that view has a similar loop like this one and could also have N+1 issues (not sure but adding a join there shouldn’t hurt too much).

Reply:
Thanks addressed it in latest commit.